frontend/src/components/Volunteer/volunteer.jsx

```diff
@@ -0,0 +1,222 @@
+import React, { Component } from "react";
+import { Container, Row, Col } from "react-bootstrap";
+import "./volunteer.css";
+import SingleCarousel from "../SingleCarousel/index";
+
+class Volunteer extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      role: "",
+      name: "",
+      email: "",
+      phone: "",
+      website: "",
+      message: ""
+    };
+  }
+
+  onNameChange = e => {
+    this.setState = {
+      name: e.target.value
+    };
+  };
+
+  onRolechange = e => {
+    this.setState = {
+      name: e.target.value
+    };
+  };
+
+  onMessageChange = e => {
+    this.setState = {
+      message: e.target.value
+    };
+  };
+
+  onWebsiteChange = e => {
+    this.setState = {
+      website: e.target.value
+    };
+  };
+
+  onEmailChange = e => {
+    this.setState = {
+      email: e.target.value
+    };
+  };
+
+  onPhoneChange = e => {
+    this.setState = {
+      phone: e.target.value
+    };
+  };
+
+  onFormSubmit = e => {
+    e.preventDefault();
+
+    let sendBackend = {
+      role: this.state.role,
+      name: this.state.name,
+      email: this.state.email,
+      phone: this.state.phone,
+      website: this.state.website,
+      message: this.state.message
+    };
+
+    this.SendData(sendBackend);
```

Comment:
`setState` is handling `state` for you so you do not need to redeclare state with `sendBackend` 

this can just be `this.SendData(this.state)`

but you dont even need sendData

I think this can be a bit cleaner as:

```
onFormSubmit = e => {
    e.preventDefault();

    fetch("http://the backend", {
      method: "POST",
      body: JSON.stringify(this.state),
      headers: { "Content-Type": "application/json"},
    }).then(function() {
      console.log("data has been set");
    });
}
```

you also are not using `onFormSubmit` in the jsx/html. Up to you if you want to include it in there.